### PR TITLE
reordering implementation before validating

### PR DESCRIPTION
**Pull request by Shreyv**
Added implementation of reordering of instantiated variables to match the order of module variables and also added valid and invalid test cases for the same.

### CODE REVIEW

**Comment by Shreyv** [comment]
@sdamghan Could you please verify this

**Comment by sdamghan** [comment]
@Shreyv Great, could you please add the same feature for tasks and functions?

**Comment by Shreyv** [comment]
@sdamghan Sure I will work on it.
Thanks

**Comment by jeanlego** [comment]
@Shreyv This looks good so far!
I would consider writing more test to cover different case, maybe empty module?
although this should not happen, odin should be able to handle it gracefully

**Comment by Shreyv** [comment]
@jeanlego  and @sdamghan Could you please verify the modifications?

Thanks

File: ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv [hunk 1]
```diff
@@ -1,120 +1,132 @@
...
+-1                , 0            , 2            , 4.3         , 0.3                , n/a                                   , 0    , -1    , -1   , 4           , 2             , 66.7            , 0   , 1.7          , timescale_syntax.blif                                      , syntax/timescale_syntax/no_arch                                                                 , -1     , 1  , -1        , 4           , 2         , 2  
+-1                , 1            , 14           , 4.3         , 0.4                , n/a                                   , 4    , -1    , -1   , 5           , 14            , 86.7            , 0   , 2.3          , bm_DL_4_bit_updown_counter.blif                            , syntax/bm_DL_4_bit_updown_counter/no_arch                                                       , -1     , 2  , -1        , 15          , 19        , 5
```
Comment (jeanlego) [inline ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv]: Can you only update the line that really changed? Also there seems to be a bug on the python script since the order is way off. what platform are you using ? I will have to test and make sure ordering is preserved...

Comment (Shreyv) [reply ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv]: I am using Ubuntu 16.04

Comment (jeanlego) [reply ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv]: pr #1245 should fix this. once merge can you regenerate the file?

Comment (Shreyv) [reply ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv]: @jeanlego Sure I will do it.

**Comment by jeanlego** [comment]
can you regenerate the results? they weren't added in

File: ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv [hunk 2]
```diff
@@ -116,5 +116,13 @@ syntax/unconnected_input/k6_frac_N10_frac_chain_mem32K_40nm
...
 syntax/undeclared_signal/no_arch                                                                , n/a                                   , undeclared_signal.blif                                     , 0   , 10.5        , 2.8          , 0.2                , 100.0           , 0            , 0  , 0  , 0            , 0    , -1   , -1        , -1    , -1     , -1                , 0           , 0           , 0             , 0         
-
-
+syntax/delay_syntax/k6_frac_N10_frac_chain_mem32K_40nm                                          , k6_frac_N10_frac_chain_mem32K_40nm.xml, delay_syntax.blif                                          , 0   , 5.3         , 17.3         , 0.2                , 100.0           , 1            , 1  , 3  , 3            , 3    , -1   , -1        , -1    , -1     , 4                 , 5           , 4           , 3             , 7
```
Comment (jeanlego) [inline ODIN_II/regression_test/benchmark/task/syntax/simulation_result.csv]: This looks better :)

**Comment by jeanlego** [comment]
@Shreyv can you resolve conflicts? I can merge after

**Comment by jeanlego** [comment]
Thanks for this
